src/snowflake/snowpark/mock/connection.py

```diff
@@ -135,6 +144,18 @@ def drop_table(self, name: Union[str, Iterable[str]]) -> None:
             if name in self.table_registry:
                 self.table_registry.pop(name)
 
+        def create_or_replace_view(
+            self, execution_plan: MockExecutionPlan, name: Union[str, Iterable[str]]
+        ):
+            name = self.get_fully_qualified_name(name)
+            self.view_registry[name] = execution_plan
```

Comment:
why we are storing the plan instead of just executing and storing the result TableEmulator like how we do with table?

Reply:
view is dynamically generated